### PR TITLE
Slim down dependencies

### PR DESCRIPTION
**Pull request by lindsayismith**
Refactor dependencies to remove a bunch of things that aren't required, and change to modern gradle dependency types

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -126,7 +126,15 @@ publishing {
     repositories {
         mavenLocal()
         maven {
-            url "https://oss.sonatype.org/service/local/staging/deploy/maven2"
+            if(project.version.endsWith('-SNAPSHOT')) {
+                url "https://us-west1-maven.pkg.dev/lindsaysmith-convoy-playground/mvn-snapshots"
```
Comment (Gridasov) [inline build.gradle:130]: Is  it an intentional change?

**Review by Gridasov — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (lindsayismith) [reply build.gradle:130]: no, thanks, I'll remove it

**Pull request merged**